SocketIOClientSwift/SocketIOClient.swift

```diff
@@ -148,28 +153,30 @@ public final class SocketIOClient: NSObject, SocketEngineClient {
     */
     public func connect(timeoutAfter timeoutAfter:Int,
         withTimeoutHandler handler:(() -> Void)?) {
-            guard status != SocketIOClientStatus.Connected else {
+            assert(timeoutAfter >= 0, "Invalid timeout: \(timeoutAfter)")
```

Comment:
Why not? It doesn't make any sense to pass a negative `timeout`. What should be the expected behavior? The `dispatch_after` would be invoked right away so it would timeout right away.